dotCMS/src/main/java/com/dotcms/contenttype/business/FieldAPIImpl.java

```diff
@@ -70,9 +140,68 @@ public FieldVariable save(FieldVariable var, User user) throws DotDataException,
 
       return fac.save(var);
   }
+
   @Override
   public void delete(Field field) throws DotDataException {
-    fac.delete(field);
+	  try {
+		  this.delete(field, this.userAPI.getSystemUser());
+	  } catch (DotSecurityException e){
+		  throw new DotDataException(e);
+	  }
+  }
+
+  @Override
+  public void delete(Field field, User user) throws DotDataException, DotSecurityException {
+
+	  ContentTypeAPI tapi = APILocator.getContentTypeAPI(user);
+	  ContentType type = tapi.find(field.contentTypeId());
+
+	  perAPI.checkPermission(type, PermissionLevel.PUBLISH, user);
+
+
+	  Structure structure = new StructureTransformer(type).asStructure();
+	  com.dotmarketing.portlets.structure.model.Field legacyField = new LegacyFieldTransformer(field).asOldField();
+
+
+      HibernateUtil.startTransaction();
```

Comment:
Should we just wrap this whole method in a local transaction?  I am asking becuase what happens if this method is getting called in another API that has already started a transaction?